lib/authorizer/edgegrid.js

```diff
@@ -0,0 +1,365 @@
+    sdk = require('postman-collection'),
+    crypto = require('crypto'),
+    querystring = require('querystring'),
+    urlEncoder = require('postman-url-encoder'),
+    RequestBody = require('postman-collection').RequestBody,
+    bodyBuilder = require('../requester/core-body-builder'),
+
+    ASCII_SOURCE = 'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789',
+    ASCII_SOURCE_LENGTH = ASCII_SOURCE.length,
+    EMPTY = '',
+    COLON = ':',
+    UTC_OFFSET = '+0000',
+    ZERO = '0',
+    DATE_TIME_SEPARATOR = 'T',
+    TAB = '\t',
+    SPACE = ' ',
+    SLASH = '/',
+    SIGNING_ALGORITHM = 'EG1-HMAC-SHA256 ',
+    AUTHORIZATION = 'Authorization',
+
+    /**
+     * Generates a random string of given length
+     *
+     * @param {Number} [length=6] Length of the string to be generated
+     * @returns {String} Randomly generated string
+     */
+    randomString = function (length) {
+        length = length || 6;
+
+        var result = [],
+            i;
+
+        for (i = 0; i < length; i++) {
+            result[i] = ASCII_SOURCE[(Math.random() * ASCII_SOURCE_LENGTH) | 0];
+        }
+
+        return result.join(EMPTY);
+    },
+
+    /**
+     * Pads the given number with '0' and returns a string with given length (or greater if number is already longer
+     * than given length)
+     *
+     * @param {Number} num Number to pad
+     * @param {Number} length Minimum length of the string to return
+     * @returns {String} Padded number
+     */
+    padNumber = function (num, length) {
+        var paddedNum = `${num}`;
+
+        while (paddedNum.length < length) {
+            paddedNum = ZERO + paddedNum;
+        }
+
+        return paddedNum;
+    },
+
+    /**
+     * Returns current timestamp in the format described in EdgeGrid specification (yyyyMMddTHH:mm:ss+0000)
+     *
+     * @returns {String} UTC timestamp in format yyyyMMddTHH:mm:ss+0000
+     */
+    getTimestamp = function () {
+        var date = new Date();
+
+        return date.getUTCFullYear() +
+                padNumber(date.getUTCMonth() + 1, 2) +
+                padNumber(date.getUTCDate(), 2) +
+                DATE_TIME_SEPARATOR +
+                padNumber(date.getUTCHours(), 2) +
+                COLON +
+                padNumber(date.getUTCMinutes(), 2) +
+                COLON +
+                padNumber(date.getUTCSeconds(), 2) +
+                UTC_OFFSET;
+    },
+
+    /**
+     * Creates a String containing a tab delimited set of headers.
+     *
+     * @param {String[]} headersToSign Headers to include in signature
+     * @param {Object} headers Request headers
+     * @returns {String} Canonicalized headers
+     */
+    canonicalizeHeaders = function (headersToSign, headers) {
+        var formattedHeaders = [],
+            headerValue;
+
+        headersToSign.forEach(function (headerName) {
+            if (typeof headerName !== 'string') { return; }
+
+            headerName = headerName.trim().toLowerCase();
```

Comment:
why trim?

Reply:
To handle user input like ' X, Y, Z ' which will be like [' X', ' Y', ' Z '] after splitting by comma.